v2/yarpcgrpc/outbound.go

```diff
@@ -252,14 +277,15 @@ func (o *Outbound) getPeerForRequest(ctx context.Context, req *yarpc.Request) (*
 		onFinish func(error)
 		err      error
 	)
-
 	if o.Chooser != nil {
 		peer, onFinish, err = o.Chooser.Choose(ctx, req)
-	} else if o.Dialer != nil && o.URL != nil {
+	} else if req.Peer != nil {
```

Comment:
It would be nice if we could generalize this choose/dialing logic, not sure if we can though 🤔